test/suites/clustering.sh

```diff
@@ -2098,28 +2100,33 @@ test_clustering_handover() {
   ns2="${prefix}2"
   spawn_lxd_and_join_cluster "${ns2}" "${bridge}" "${cert}" 2 1 "${LXD_TWO_DIR}"
 
+  echo "Launched member 2"
+
   # Spawn a third node
   setup_clustering_netns 3
   LXD_THREE_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
   chmod +x "${LXD_THREE_DIR}"
   ns3="${prefix}3"
   spawn_lxd_and_join_cluster "${ns3}" "${bridge}" "${cert}" 3 1 "${LXD_THREE_DIR}"
 
+  echo "Launched member 3"
+
   # Spawn a fourth node, this will be a non-voter, stand-by node.
   setup_clustering_netns 4
   LXD_FOUR_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
   chmod +x "${LXD_FOUR_DIR}"
   ns4="${prefix}4"
   spawn_lxd_and_join_cluster "${ns4}" "${bridge}" "${cert}" 4 1 "${LXD_FOUR_DIR}"
 
+  echo "Launched member 4"
+
   LXD_DIR="${LXD_TWO_DIR}" lxc cluster list
   LXD_DIR="${LXD_TWO_DIR}" lxc cluster show node4 | grep -q "\- database-standby"
 
   # Shutdown the first node.
   LXD_DIR="${LXD_ONE_DIR}" lxd shutdown
 
-  # Wait some time to possibly allow for a leadership change.
-  sleep 10
```

Comment:
Does `lxd shutdown` wait for the leadership transfer to complete? In that case the sleep should not be needed indeed.

Reply:
Yes, well at least I think it does, here https://github.com/lxc/lxd/blob/master/lxd/daemon.go#L1490-L1498 (this was moved to earlier in the shutdown sequence recently).

I still see this part of the test failing sometimes (both with and without a sleep after lxd shutdown):

https://github.com/lxc/lxd/blob/cd68c7db7fd03ca949aa6210058b1292e362b695/test/suites/clustering.sh#L2137

I'm not sure how long we should expect have to wait for the new leader to consider the stopped member no longer a database-standby role.

Reply:
The role change is performed by LXD, not sure when it triggers.

Reply:
I'll keep running the test over at https://github.com/lxc/lxd/pull/9296 and I'll show you what I mean when it fails.
The role is handed over when LXD shutsdown, but the new leader considers the server that is stopping to be a database-standby (which AFAIK is expected because it handed over leadership) until such time as it has shutdown at somepoint it is detected and becomes not a database role at all. Most of the time this is immediate but sometimes not.

Reply:
I'll try to reproduce too in meanwhile.